RebornCore/src/main/java/reborncore/common/blockentity/SlotConfiguration.java

```diff
@@ -291,53 +289,14 @@ private void handleItemInput(MachineBaseBlockEntity machineBase) {
 			if (targetStack.getMaxCount() == targetStack.getCount()) {
 				return;
 			}
-			Inventory sourceInv = InventoryUtils.getInventoryAt(machineBase.getWorld(), machineBase.getPos().offset(side));
-			if (sourceInv == null) {
-				return;
-			}
-
-			IntList availableSlots = null;
 
-			if (sourceInv instanceof SidedInventory) {
-				availableSlots = IntArrayList.wrap(((SidedInventory) sourceInv).getAvailableSlots(side.getOpposite()));
-			}
-
-			for (int i = 0; i < sourceInv.size(); i++) {
-				if (availableSlots != null && !availableSlots.contains(i)) {
-					continue;
-				}
-
-				ItemStack sourceStack = sourceInv.getStack(i);
-				if (sourceStack.isEmpty()) {
-					continue;
-				}
-				if(!canInsertItem(slotID, sourceStack, side, machineBase)){
-					continue;
-				}
-
-				if (sourceInv instanceof SidedInventory && !((SidedInventory) sourceInv).canExtract(i, sourceStack, side.getOpposite())) {
-					continue;
-				}
-
-				//Checks if we are going to merge stacks that the items are the same
-				if (!targetStack.isEmpty()) {
-					if (!ItemUtils.isItemEqual(sourceStack, targetStack, true, false)) {
-						continue;
-					}
-				}
-				int extract = 4;
-				if (!targetStack.isEmpty()) {
-					extract = Math.min(targetStack.getMaxCount() - targetStack.getCount(), extract);
-				}
-				ItemStack extractedStack = sourceInv.removeStack(i, extract);
-				if (targetStack.isEmpty()) {
-					inventory.setStack(slotID, extractedStack);
-				} else {
-					inventory.getStack(slotID).increment(extractedStack.getCount());
-				}
-				inventory.setHashChanged();
-				break;
-			}
+			StorageUtil.move(
+					ItemStorage.SIDED.find(machineBase.getWorld(), machineBase.getPos().offset(side), side.getOpposite()),
+					InventoryStorage.of(machineBase, null).getSlot(slotID),
+					iv -> true,
+					Long.MAX_VALUE,
```

Comment:
This was 4 before, I think this should be kept the same speed?

Reply:
Fixed.